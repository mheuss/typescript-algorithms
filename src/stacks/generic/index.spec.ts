import { Stack } from "./index";

describe("Test of Generic Stack", () => {
  const stack = new Stack<number>();

  it("Should be able to push items on it", () => {
    stack.put(1);
    stack.put(2);
    stack.put(3);
    stack.put(4);
    stack.put(5);
    stack.put(6);
    stack.put(7);
    expect(stack.toArray()).toMatchSnapshot();
  });

  it("Should be able to tell us the length", () => {
    expect(stack.length()).toEqual(7);
  });

  it("Should be able to push items off", () => {
    expect(stack.get()).toEqual(7);
    expect(stack.get()).toEqual(6);
    expect(stack.get()).toEqual(5);
    expect(stack.get()).toEqual(4);
    expect(stack.get()).toEqual(3);
    expect(stack.get()).toEqual(2);
    expect(stack.get()).toEqual(1);
  });

  it("Should return nothing if there is nothing to get", () => {
    expect(stack.get()).toBeUndefined();
  });

  it("Should be able to empty the array", () => {
    stack.discard();
    stack.put(1);
    stack.put(2);
    stack.put(3);
    stack.put(4);
    stack.put(5);
    stack.put(6);
    stack.put(7);
    expect(stack.length()).toEqual(7);
  });

  it("Should be able to export an array and load it back in with the same order", () => {
    stack.discard();
    stack.put(1);
    stack.put(2);
    stack.put(3);
    stack.put(4);
    stack.put(5);
    stack.put(6);
    stack.put(7);
    const arr = stack.toArray();
    expect(arr).toMatchSnapshot();
    stack.discard();

    const QUEUE = "queue";
    stack.load(arr);
    expect(stack[QUEUE]).toMatchSnapshot();
  });
});
