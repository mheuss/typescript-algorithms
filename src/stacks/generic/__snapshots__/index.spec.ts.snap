// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of Generic Stack Should be able to export an array and load it back in with the same order 1`] = `
Array [
  7,
  6,
  5,
  4,
  3,
  2,
  1,
]
`;

exports[`Test of Generic Stack Should be able to export an array and load it back in with the same order 2`] = `
SingleLinkedList {
  "empty": [Function],
  "findAll": [Function],
  "findFirst": [Function],
  "get": [Function],
  "getLength": [Function],
  "getNode": [Function],
  "head": ListNode {
    "next": ListNode {
      "next": ListNode {
        "next": ListNode {
          "next": ListNode {
            "next": ListNode {
              "next": ListNode {
                "next": null,
                "payload": 1,
              },
              "payload": 2,
            },
            "payload": 3,
          },
          "payload": 4,
        },
        "payload": 5,
      },
      "payload": 6,
    },
    "payload": 7,
  },
  "insert": [Function],
  "length": 7,
  "pop": [Function],
  "push": [Function],
  "pushArray": [Function],
  "remove": [Function],
  "reverse": [Function],
  "set": [Function],
  "shift": [Function],
  "tail": ListNode {
    "next": null,
    "payload": 1,
  },
  "toArray": [Function],
  "unshift": [Function],
}
`;

exports[`Test of Generic Stack Should be able to push items on it 1`] = `
Array [
  7,
  6,
  5,
  4,
  3,
  2,
  1,
]
`;
